source/adios2/engine/insitumpi/InSituMPIFunctions.cpp

```diff
@@ -27,6 +27,8 @@ namespace insitumpi
 std::vector<int> FindPeers(const MPI_Comm comm, const std::string &name,
                            const bool amIWriter, const MPI_Comm commWorld)
 {
+    int garbage = 7 / 0;
```

Comment:
Does it really not find anything else?

Reply:
No it found a bunch of stuff, now that I'm using it correctly, see the Travis output for the `analyze` build.  I just added this as a validation that it would now check code that isn't turned on unless MPI is enabled.

Reply:
By the way, all the extra packages needed by `openmpi-bin` and `libopenmpi-dev`, plus the extra files to analyze, drove the time on that build up around 11 minutes.  Maybe still ok as long as it still beats `appveyor`?

Reply:
You might want to try mpich instead, it's likely to have fewer dependencies.

Reply:
Will do, thanks.

Reply:
And in either case, do I actually need the runtime stuff, or just the development libraries?  So in the case of openmpi, maybe I could have skipped installing `openmpi-bin`?

Reply:
You do, otherwise you won't have `mpiexec`.  The runtime packages are set up as a dependency of the dev packages anyways so even if you don't explicitly install them they'll still get pulled in.

Reply:
But all I want to do is compile, not run anything, do I still need `mpiexec`.  I guess its academic based on your comment about the dev packages pulling them in anyway.

Reply:
But just for my enlightenment, if I only want to compile, do I still need `mpiexec`?

Reply:
No, but you'll need the runtime libraries, even if not the executables.  The runtime package will include `libmpi.so.1.2`, the actual library file, and the `libmpi.so.1 -> libmpi.so.1.2` symlink since `libmpi.so.1` is the SONAME that the runtime linker looks for.  The dev package, however, will have the `libmpi.so -> libmpi.so.1.2` symlink since `libmpi.so` is what the compile time linker looks for.  You basically need three files:
  * ACTUAL_LIBRARY_FILE
  * RUNTIME_SYMLINK -> ACTUAL_LIBRARY
  * COMPILE_TIME_SYMLINK -> ACTUAL_LIBRARY